test/DaedalusCompiler.Tests/ParsingSourceToAbstractAssemblyTests.cs

```diff
@@ -5828,5 +5889,181 @@ class Person {
+            _instructions = GetExecBlockInstructions("Mage");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Archer");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Barbarian");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Orc");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Warrior");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Call(Ref("Orc")),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Elite");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Call(Ref("Orc")),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Shaman");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Call(Ref("Orc")),
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+
+            
+            _expectedSymbols = new List<DatSymbol>
+            {
+                Ref("C_NPC"),
+                Ref("C_NPC.data"),
+                
+                Ref("Mage"),
+                Ref("Archer"),
+                Ref("Barbarian"),
+                
+                Ref("Orc"),
+                
+                Ref("Warrior"),
+                Ref("Elite"),
+                Ref("Shaman")
+            };
+            AssertSymbolsMatch();
+        }
+        
+        
+        [Fact]
+        public void TestIntArrayLocalVariableReturn()
+        {
+            _code = @"
+                func int testFunc() {
+                    var int a[2048];
+                    return a;
+                };
+
+                func int secondFunc() {
+                    const int b[3] = {1, 2, 3};
+                    return b;
+                };
+           ";
+
+            
+            _instructions = GetExecBlockInstructions("testFunc");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new PushVar(Ref("testFunc.a")),
+                new Ret(),
+                
+                new Ret(),
```

Comment:
Why `Ret` 2 times?

Reply:
Look at other tests, it's always like that. There is always implicitly added `Ret` instruction at the end of a function and it doesnt matter if you have `return` in your code (in this case there will be 2 Ret instructions) or not.